cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -61,13 +61,18 @@ void sc_init_invocation(sc_invocation *inv, const struct sc_args *args, const ch
         die("cannot run with NULL executable");
     }
 
+    /* Instance name length + 1 overflow + 1 NULL */
```

Comment:
This entire part about invocation knowing the instance name vs snap name could be merged separately as well if you want to. It's a relatively painless +1 and reduces this branch further to the essence of the problem.

Reply:
Opened as #7406

Reply:
> Opened as #7406

@bboozzoo - please see my comments in 7406.